File: src/main/java/com/dg/watcher/history/graph/HistoryGraphTooltipGenerator.java
package com.dg.watcher.history.graph;

import org.jfree.chart.labels.CategoryToolTipGenerator;
import org.jfree.data.category.CategoryDataset;
import static com.dg.watcher.base.Const.GRAPH_TOOLTIP;
import static java.lang.String.format;
import static java.util.Locale.ENGLISH;


public class HistoryGraphTooltipGenerator implements CategoryToolTipGenerator {
    @Override
    public String generateToolTip(CategoryDataset categoryDataset, int series, int itemIndex) {
        String buildName = categoryDataset.getColumnKey(itemIndex).toString();
        float  apkSizeInMb = categoryDataset.getValue(series, itemIndex).floatValue();

        return format(ENGLISH, GRAPH_TOOLTIP, buildName, apkSizeInMb);
    }
}

File: src/main/java/com/dg/watcher/history/graph/HistoryGraphTooltipGenerator.kt
package com.dg.watcher.history.graph

import org.jfree.chart.labels.CategoryToolTipGenerator
import org.jfree.data.category.CategoryDataset
import com.dg.watcher.base.GRAPH_TOOLTIP
import java.lang.String.format
import java.util.Locale.ENGLISH


class HistoryGraphTooltipGenerator : CategoryToolTipGenerator {
    override fun generateToolTip(categoryDataset: CategoryDataset, series: Int, itemIndex: Int): String {
        val buildName = categoryDataset.getColumnKey(itemIndex).toString()
        val apkSizeInMb = categoryDataset.getValue(series, itemIndex).toFloat()

        return format(ENGLISH, GRAPH_TOOLTIP, buildName, apkSizeInMb)
    }
}